src/overlays/actors/ovl_Boss_03/z_boss_03.c

```diff
@@ -41,157 +269,2320 @@ const ActorInit Boss_03_InitVars = {
+    { 0.0f, 0.0f, 0.0f },         { 0.0f, 0.0f, 0.0f },          { 0.0f, 0.0f, 0.0f },
+    { 0.0f, 0.0f, 0.0f },
 };
 
-// static ColliderJntSphInit sJntSphInit = {
-static ColliderJntSphInit D_809E8FD8 = {
-    { COLTYPE_METAL, AT_ON | AT_TYPE_ENEMY, AC_ON | AC_TYPE_PLAYER, OC1_ON | OC1_TYPE_PLAYER, OC2_TYPE_1, COLSHAPE_JNTSPH, },
-    5, D_809E8F24, // sJntSphElementsInit,
+Color_RGBA8 sGyorgDustPrimColor = { 60, 50, 20, 255 };
+Color_RGBA8 sGyorgDustEnvColor = { 40, 30, 30, 255 };
+
+/**
+ * Used when chasing Player and Gyorg is near the bottom
+ */
+void Boss03_SpawnDust(Boss03* this, GlobalContext* globalCtx) {
+    if (this->insideJawPos.y < 80.0f) {
+        u8 i;
+        Vec3f pos;
+        Vec3f velocity;
+        Vec3f accel;
+
+        for (i = 0; i < 5; i++) {
+            velocity.x = randPlusMinusPoint5Scaled(10.0f);
+            velocity.y = Rand_ZeroFloat(2.0f) + 2.0f;
+            velocity.z = randPlusMinusPoint5Scaled(10.0f);
+
+            accel.y = -0.075f;
+            accel.z = 0.0f;
+            accel.x = 0.0f;
+
+            pos.y = Rand_ZeroFloat(20.0f) + 5.0f;
+            pos.z = randPlusMinusPoint5Scaled(150.0f) + this->insideJawPos.z;
+            pos.x = randPlusMinusPoint5Scaled(150.0f) + this->insideJawPos.x;
+
+            func_800B0EB0(globalCtx, &pos, &velocity, &accel, &sGyorgDustPrimColor, &sGyorgDustEnvColor,
+                          Rand_ZeroFloat(200.0f) + 400.0f, 10, Rand_ZeroFloat(10.0f) + 25.0f);
+        }
+    }
+}
+
+void Boss03_Init(Actor* thisx, GlobalContext* globalCtx2) {
+    Boss03* this = THIS;
+    s32 i;
+    GlobalContext* globalCtx = globalCtx2;
+    Vec3f sp70;
+
+    if (gSaveContext.weekEventReg[55] & 0x80) {
+        Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_DOOR_WARP1, 0.0f, PLATFORM_HEIGHT,
+                           200.0f, 0, 0, 0, ENDOORWARP1_FF_1);
+        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_ITEM_B_HEART, 0.0f, PLATFORM_HEIGHT, 0.0f, 0, 0, 0, 0);
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+
+    if (this->actor.params == GYORG_PARAM_SEAWEED) {
+        this->actor.update = Boss03_SeaweedUpdate;
+        this->actor.draw = Boss03_SeaweedDraw;
+
+        this->unk_240 = Rand_ZeroFloat(1000.0f);
+
+        this->actor.scale.x = Rand_ZeroFloat(0.0075f) + 0.027f;
+        this->actor.scale.y = 0.02f;
+        this->actor.scale.z = 0.015f;
+
+        // Manually set the joint table of seaweed
+        for (i = 0; i < 6; i++) {
+            this->jointTable[i].x = Math_SinS(this->unk_240 * 0x100 + i * 15000) * 3000.0f;
+            this->jointTable[i].y = Math_SinS(this->unk_240 * 0x180 + i * 20000) * 2000.0f;
+            this->jointTable[i].z = Math_SinS(this->unk_240 * 0x10 + i * 19000) * 4000.0f;
+        }
+
+        this->actor.flags &= ~ACTOR_FLAG_1;
+        return;
+    }
+
+    this->actor.world.pos = sGyorgInitialPos;
+
+    Boss03_SeedRand(1, 29093, 9786);
```

Comment:
You could note that this is actually deterministic since it's only used in this function.